framework/base/Component.php

```diff
@@ -475,12 +474,18 @@ public function hasEventHandlers($name)
      */
     public function on($name, $handler, $data = null, $append = true)
     {
+		$priority = (int) $append;
         $this->ensureBehaviors();
-        if ($append || empty($this->_events[$name])) {
-            $this->_events[$name][] = [$handler, $data];
-        } else {
-            array_unshift($this->_events[$name], [$handler, $data]);
-        }
+
+		if (empty($this->_events[$name])) {
+			$this->_events[$name] = new EventPriorityQueue();
```

Comment:
I am of the opinion that is not very safe for this "Queue" but I can understand this is a Yii2 approach. I ll commit that.